metaflow/metadata/metadata.py

```diff
@@ -672,6 +672,22 @@ def _register_system_metadata(self, run_id, step_name, task_id, attempt):
         if metadata:
             self.register_metadata(run_id, step_name, task_id, metadata)
 
+    @classmethod
+    def _filter_tasks_by_metadata(
+        cls, flow_id, run_id, query_step, field_name, field_value
+    ):
+        raise NotImplementedError()
+
+    @classmethod
+    def filter_tasks_by_metadata(
+        cls, flow_id, run_id, query_step, field_name, field_value
+    ):
+        # TODO: Do we need to do anything wrt to task attempt?
+        task_ids = cls._filter_tasks_by_metadata(
+            flow_id, run_id, query_step, field_name, field_value
+        )
+        return task_ids
+
```

Comment:
is there a need for the private method, or could this simply be contained in the public-facing one? right now its not doing anything before calling the private one.

Reply:
also, did you have an implementation of this for `service.py` yet?